File: Portal/ClientApp/src/app/_autogenerated/itemBindingModel.ts

//----------------------------------------------
//  DTO interface and model
//----------------------------------------------

export interface IItemBindingModel  {
    name: string;
    email: string;
    from: Date;
    to: Date;
    isActive: boolean;
    optionId: number;
    categoryId: number;
    parts: number[];
}

export class ItemBindingModel  implements IItemBindingModel{ 
    public name: string;
    public email: string;
    public from: Date;
    public to: Date;
    public isActive: boolean;
    public optionId: number;
    public categoryId: number;
    public parts: number[];

    constructor(name : string = null,  email : string = null,  from : Date = null,  to : Date = null,  isActive : boolean = null,  optionId : number = null,  categoryId : number = null,  parts : number[] = null ) {
        
        this.name = name;
        this.email = email;
        this.from = from;
        this.to = to;
        this.isActive = isActive;
        this.optionId = optionId;
        this.categoryId = categoryId;
        this.parts = parts;
    }
}

File: Portal/Models/BindingModel/ItemBindingModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Angular.Core.Models.BindingModel
{
    public class ItemBindingModel
    {
        [Required]
        public string Name { get; set; }
        [Required]
        [EmailAddress]
        public string Email { get; set; }
        public DateTime From { get; set; }
        public DateTime To { get; set; }
        public bool IsActive { get; set; }
        public int OptionId { get; set; }
        public int? CategoryId { get; set; }
        public IEnumerable<int> Parts { get; set; }

        public override string ToString() => $"{Name}, {Email}";

    }
}
